src/st/no0/e_stone_rose.c

```diff
@@ -24,8 +24,104 @@ Entity* func_us_801D7D00(u16 arg0) {
     return NULL;
 }
 
+extern PfnEntityUpdate D_us_8018091C[];
+extern s16 D_us_80182268[][4];
```

Comment:
Seems likely this is a struct array?
```
typedef struct {
    s16 unk0;
    s16 unk2;
    s16 unk4;
    s16 unk6;
} Unk_D_us_80182268;

extern Unk_D_us_80182268 D_us_80182268[];
```

Then
```
        self->rotX = D_us_80182268[params].unk0 & 0xFF;
        self->rotY = D_us_80182268[params].unk2 & 0xFF;
        self->rotZ = D_us_80182268[params].unk4 & 0x7F;
```
etc.?

Reply:
I'm actually not sure if this is better than not since the data is packed. e.g. unk0 is both rotX and the upper part is hitEffect, so I don't think it can be given a sane name. rotXAndHitEffect? I'd probably get Xeeynamo's opinion before changing it again though